File: TYPESCRIPT/Cap08/c08ex09.ts
// c08ex09.ts

import * as rls from 'readline-sync';

function Fibonacci(N: number): number {
  if (N == 0) {
    return 0;
  }
  if (N == 1) {
    return 1;
  }
  if (N >= 2) {
    return Fibonacci(N - 1) + Fibonacci(N - 2);
  }
  return 0; // valor padrão para evitar erros de compilação
}

let I: number;

for (I = 0; I <= 14; I++) {
  console.log(`${I < 10 ? ` ${I}` : I} - ${Fibonacci(I)}`);
}

console.log("");
process.stdout.write("Tecle <Enter> para encerrar... ");
rls.question();

File: JULIA/Cap08/c08ex09.jl
# c08ex09.jl

using Printf

function fibonacci(n)::Int64
  if (n == 0)
    return 0
  end
  if (n == 1)
    return 1
  end
  if (n >= 2)
    return fibonacci(n - 1) + fibonacci(n - 2)
  end
end

for i in 0:14
  @printf "%2d - %d\n" i fibonacci(i)
end

print("\nTecle <Enter> para encerrar... ")
readline()
